mongo/client.go

```diff
@@ -613,7 +613,15 @@ func (c *Client) configure(opts *options.ClientOptions) error {
 		if err := opts.ServerAPIOptions.ServerAPIVersion.Validate(); err != nil {
 			return err
 		}
-		c.serverAPI = *opts.ServerAPIOptions
+
+		// manually clone the passed in options to avoid future modifications of original ServerAPIOptions.
+		c.serverAPI = options.ServerAPI().SetServerAPIVersion(opts.ServerAPIOptions.ServerAPIVersion)
```

Comment:
Nit: assigning the result to `c.serverAPI` is unnecessary since `SetServerAPIVersion` modifies the struct. The setters return the receiver as a convenience to chain calls. So I'd suggest removing it, since it makes it seem the assignment is necessary.

Reply:
Removed! Wasn't sure what to do with the return value.